### PR TITLE
Fix Issue 17870 - Can't alias a mix of parent and child class members

### PR DESCRIPTION
**Pull request by BorisCarvajal**

### CODE REVIEW

**Review by ghost — APPROVED** [review]

File: test/compilable/test17870.d [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+class A
+{
+    int a = 1;
+}
+
+class B : A
+{
+    int b = 2;
+    alias tup = AliasSeq!(b, a);
+}
```
Comment (wilzbach) [inline test/compilable/test17870.d:12]: What if there's more nesting, i.e.

class A0 
{
   int z = 3;
}

class A : A0

static assert(ins.tup == AliasSeq!(2,1, 3));
...


```

**Review by wilzbach — COMMENTED** [review]
I think we should add a test for more complex class hierarchies (at least more than father-child).

**Review by RazvanN7 — APPROVED** [review]

**Comment by RazvanN7** [comment]
@BorisCarvajal Please add the test requested by @wilzbach and we are good to go.

**Comment by RazvanN7** [comment]
Closed in favor of #13380

**Pull request closed without merging**